pages/blog/the-anatomy-of-a-graphql-request.mdx

```diff
@@ -0,0 +1,341 @@
+
+For the following invalid GraphQL operation (`)` missing after `$id`):
+
+```
+query UserById($id: ID!) {
+  user(id: $id {
+    id
+    name
+  }
+}
+```
+
+The error will look similar to this:
+
+```json
+{
+  "message": "Syntax Error: Expected Name, found {",
+  "locations": [
+    {
+      "line": 2,
+      "column": 16
+    }
+  ]
+}
+```
+
+As you can, see the error messages are unfortunately not always straight-forward and helpful. The location can, however, help you tracking down the syntax error!
+
+In case no error occurs, the parse phase produces an AST (abstract syntax tree).
+
+The AST is a handy format that is used for the follow-up phases `validate` and `execute`.
+
+For our operation it would be identical to the following JSON:
+
+```json
+{
+  "kind": "Document",
+  "definitions": [
+    {
+      "kind": "OperationDefinition",
+      "operation": "query",
+      "name": {
+        "kind": "Name",
+        "value": "UserById"
+      },
+      "variableDefinitions": [
+        {
+          "kind": "VariableDefinition",
+          "variable": {
+            "kind": "Variable",
+            "name": {
+              "kind": "Name",
+              "value": "id"
+            }
+          },
+          "type": {
+            "kind": "NonNullType",
+            "type": {
+              "kind": "NamedType",
+              "name": {
+                "kind": "Name",
+                "value": "ID"
+              }
+            }
+          }
+        }
+      ],
+      "selectionSet": {
+        "kind": "SelectionSet",
+        "selections": [
+          {
+            "kind": "Field",
+            "name": {
+              "kind": "Name",
+              "value": "user"
+            },
+            "arguments": [
+              {
+                "kind": "Argument",
+                "name": {
+                  "kind": "Name",
+                  "value": "id"
+                },
+                "value": {
+                  "kind": "Variable",
+                  "name": {
+                    "kind": "Name",
+                    "value": "id"
+                  }
+                }
+              }
+            ],
+            "selectionSet": {
+              "kind": "SelectionSet",
+              "selections": [
+                {
+                  "kind": "Field",
+                  "name": {
+                    "kind": "Name",
+                    "value": "id"
+                  }
+                },
+                {
+                  "kind": "Field",
+                  "name": {
+                    "kind": "Name",
+                    "value": "name"
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+This procedure is performed by the `parse` function that is exported from `graphql-js`. Other languages that are orientate themselves on the `graphql-js` reference implementation have a corresponding counterpart.
```

Comment:
> Other languages that are orientate themselves on the `graphql-js` reference implementation have a corresponding counterpart.

I am not sure what you meant here 🤔 
Do you mean that other language-based GraphQL implementations (ex: Ruby) do follow the GraphQL specification?

Reply:
I meant that other languages have also a Parser implementation and AST structure for implementing this. However, when u use for example ruby or GraphQL dot net it won't be a JSON object, only when serialized - does that make sense? Do you have an alternative formulation for this?